src/cron/updateCurrentStreak.ts

```diff
@@ -16,17 +21,43 @@ const cron: Cron = {
           )
           .addSelect('us.currentStreak', 'current')
           .addSelect('u."weekStart"', 'weekStart')
+          .addSelect(
+            `(date_trunc('day', usa."lastRecoverAt" at time zone COALESCE(u.timezone, 'utc'))::date) - interval '1 day'`,
+            'lastRecoverAt',
+          )
           .from(UserStreak, 'us')
           .innerJoin(User, 'u', 'u.id = us."userId"')
+          .leftJoin(
```

Comment:
This whole query is just something we need to closely monitor for performance/optimizations (no direct action needed)

Reply:
Agreed. This may run in the bg but still, we should keep an eye out if it behaves weirdly after this.